src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,118 @@
-import React from 'react';
+import React, { FormEvent, useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { addComment } from '../api/comments';
+import { CommentFormErrors, EMAIL_REGEXP, ErrorTypes } from '../constants';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { ErrorIcon } from './ErrorIcon';
+
+type Props = {
+  post: Post | null;
+  setErrorType: React.Dispatch<React.SetStateAction<ErrorTypes | null>>;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  post,
+  setComments,
+  setErrorType,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [comment, setComment] = useState('');
```

Comment:
I'd recommend you to create one state for your form values (use the object `{name: '', email: '', comment: ''}`